### PR TITLE
[812E] Address unflattened content with components which file name is not the GUID

### PR DESCRIPTION
**Pull request by jvega190**
https://github.com/craftersoftware/craftercms/issues/812

### CODE REVIEW

**Comment by jvega190** [comment]
Left as draft, there's still some work pending

File: ui/app/src/components/LegacyFormDialog/EmbeddedLegacyContainer.tsx
Comment (rart) [inline ui/app/src/components/LegacyFormDialog/EmbeddedLegacyContainer.tsx:1]: Add changes to changelog. 
Also, bring the 4.1.x branch content in too.

File: ui/app/src/components/PreviewConcierge/PreviewConcierge.tsx [hunk 1]
```diff
@@ -630,66 +637,66 @@ export function PreviewConcierge(props: PropsWithChildren<{}>) {
...
 
-            if (payload.documentDomain) {
-              try {
-                document.domain = payload.documentDomain;
-              } catch (e) {
-                console.error(e);
-              }
-            } else if (document.domain !== originalDocDomain) {
-              document.domain = originalDocDomain;
+          // TODO: Remove *all* document domain related stuff for `develop`
```
Comment (rart) [inline ui/app/src/components/PreviewConcierge/PreviewConcierge.tsx]: Not done.

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/app/src/CHANGELOG.md [hunk 2]
```diff
@@ -14,6 +14,7 @@
     * `LegacyRecentActivityDashlet`
     * `LegacyRecentlyPublishedDashlet`
     * `LegacyUnpublishedDashlet`
+  * Removed `documentDomain` property from ExperienceBuilder component and its usage in `PreviewConcierge` component
```
Comment (rart) [inline ui/app/src/CHANGELOG.md:17]: Missing renaming of updateEditDialogConfig and removal of document domain

**Review by rart — CHANGES_REQUESTED** [review]

**Review by rart — APPROVED** [review]

**Pull request merged**